File: src/Tests/Behavioral/PackageVarInitOrder/registry.cs
namespace go;

partial class main_package {

[GoType] partial struct reg {
    internal slice<@string> entries;
    internal nint count;
}

internal static ж<reg> newReg() {
    return Ꮡ(new reg(nil));
}

[GoRecv] internal static @string add(this ref reg r, @string name) {
    r.entries = append(r.entries, name);
    r.count++;
    return name + "-added"u8;
}

internal static ж<reg> registry = newReg();

internal static slice<@string> names = new @string[]{"stdin"u8, "stdout"u8}.slice();

[GoType("num:uint8")] partial struct kind;

internal static readonly kind kindNone = /* iota */ 0;
internal static readonly kind kindFile = 1;
internal static readonly kind kindPipe = 2;

[GoType("@string")] partial struct label;

internal static readonly label labelPipe = "pipe"u8;

} // end main_package

File: src/Tests/Behavioral/PackageVarInitOrder/registry.go
package main

type reg struct {
	entries []string
	count   int
}

func newReg() *reg {
	return &reg{}
}

func (r *reg) add(name string) string {
	r.entries = append(r.entries, name)
	r.count++
	return name + "-added"
}

var registry = newReg()

var names = []string{"stdin", "stdout"}

// kind is a NAMED integer type, so its constants have no C# `const` form (a [GoType] wrapper
// struct cannot be declared const, CS0283) and are emitted as `static readonly` FIELDS.
type kind uint8

const (
	kindNone kind = iota
	kindFile
	kindPipe
)

// A named-STRING const is the same shape (@string is a struct).
type label string

const labelPipe label = "pipe"
